src/wire/ipv6hbh.rs

```diff
@@ -105,6 +106,18 @@ impl<'a> Repr<'a> {
             buffer = &mut buffer[opt.buffer_len()..];
         }
     }
+
+    /// The hop-by-hop header containing a PadN and a MLDv2 router alert option.
+    pub fn mldv2_router_alert(n: u8) -> Self {
```

Comment:
Shouldn't the padding be calculated at the end, when emitting the options?

Reply:
To keep the API surface small, I opted to let `mldv2_router_alert()` handle both options. But I agree that going forward it may be good to have a dedicated function to emit the padding option. I have pushed a commit addressing that.